com.blackmagicdesign.Resolve.yml

```diff
@@ -0,0 +1,186 @@
+                Categories=AudioVideo
+                EOF
+                cp -p squashfs-root/graphics/blackmagicraw-speedtest_256x256_apps.png ${PREFIX}/share/icons/hicolor/256x256/apps/${APP_ID}.RAWSpeedTest.png
+            fi
+          - |
+            cat <<EOF > ${PREFIX}/share/applications/${APP_ID}.desktop
+            [Desktop Entry]
+            Version=1.0
+            Type=Application
+            Name=${APP_DESCRIPTION}
+            Name[en_US]=${APP_DESCRIPTION}
+            GenericName=${APP_DESCRIPTION}
+            Comment=Revolutionary new tools for editing, visual effects, color correction and professional audio post production, all in a single application!
+            Exec=/app/bin/resolve.sh %U
+            Icon=${APP_ID}
+            Terminal=false
+            MimeType=application/x-resolveproj;
+            StartupNotify=true
+            Categories=AudioVideo
+            EOF
+          - cp -rp squashfs-root/graphics/DV_Resolve.png ${PREFIX}/share/icons/hicolor/128x128/apps/${APP_ID}.png
+          - |
+            if [[ -e "${PREFIX}/DaVinci Control Panels Setup/DaVinci Control Panels Setup" ]]; then
+                cat <<'    EOF'| sed -r 's/^\s+//' > ${PREFIX}/share/applications/${APP_ID}.PanelSetup.desktop
+                [Desktop Entry]
+                Version=1.0
+                Encoding=UTF-8
+                Type=Application
+                Name=DaVinci Resolve Panels Setup
+                Exec="/app/DaVinci Resolve Panels Setup/DaVinci Resolve Panels Setup"
+                Icon=${APP_ID}.PanelSetup.png
+                Terminal=false
+                StartupNotify=true
+                Categories=AudioVideo
+                EOF
+                cp -rp squashfs-root/graphics/DV_Panels.png ${PREFIX}/share/icons/hicolor/128x128/apps/${APP_ID}.PanelSetup.png
+            fi
+          - |
+            if [[ -e "${PREFIX}/bin/DaVinci Remote Monitoring" ]]; then
+                cat <<'    EOF'| sed -r 's/^\s+//' > ${PREFIX}/share/applications/${APP_ID}.RemoteMonitoring.desktop
+                [Desktop Entry]
+                Version=1.0
+                Encoding=UTF-8
+                Type=Application
+                Name=DaVinci Remote Monitoring
+                Exec="/app/bin/DaVinci Remote Monitoring"
+                Icon=${APP_ID}.RemoteMonitoring
+                Terminal=false
+                StartupNotify=true
+                Categories=AudioVideo
+                EOF
+                cp -rp squashfs-root/graphics/Remote_Monitoring.png ${PREFIX}/share/icons/hicolor/128x128/apps/${APP_ID}.RemoteMonitoring.png
+            fi
+
+      - type: file
+        path: com.blackmagicdesign.Resolve.desktop
+      - type: file
+        path: com.blackmagicdesign.Resolve.png
+      - type: file
+        path: com.blackmagicdesign.Resolve.appdata.xml
+      - type: file
+        url: https://github.com/jolygmanka/filehost/releases/download/addfile2/Davinci_Resolve_19.0.1_Linux_part_aa
```

Comment:
what is this? AFAIK DaVinci NOT redistributable amd this is coming from some "unknown" source.... doesn't look official.

Reply:
Given that trying to download at the source gives this:

![image](https://github.com/user-attachments/assets/36013532-3498-4946-b340-beb380677b05)

Reply:
Cannot be downloaded officially

Reply:
then this cannot be used. If it could it would have to use `extra-data` anyway as it is required for non redistributable.

But right now there is now way to legally comply with what you are doing here.

Reply:
Yea this won't work, we cannot host illegal distributions even with extra-data

Reply:
I have a script that retrieves the latest version of a specified application DaVinci Resolve from Blackmagic Design’s API by generating URLs and headers for a curl request. However, this results in an indirect download link that requires authorization from the build system to access the internet, which doesn’t work in --sandbox mode.

Reply:
That doesn't make any more legal. Regardless the use 'extra-data' is required here. Which doesn't solve the requirements from the vendor.

Reply:
How can this problem be solved?

Reply:
maybe you misunderstood me, you can download officially through the official api, using curl requests and filling in headers and cookies, but as I understand it, flathub only allows direct links to the file in sources

Reply:
I can take it directly with DaVinci. They are the one making this difficult.

Reply:
> you can download officially through the official api, using curl requests and filling in headers and cookies,

In some jurisdiction this is consider as computer law per statute. Also the vendor could decide it's copyright infringement.

1. not using extra-data is NOT an option here. Whether you like it or not.
2. that other part legally suspicious.

Again if you see 2. as a problem take it directly with the vendor.